src/org/labkey/test/components/glassLibrary/grids/ResponsiveGrid.java

```diff
@@ -447,44 +491,39 @@ protected final WebElement getColumnHeaderCell(String headerText)
         }
 
         private List<String> columnNames;
-        protected List<String> getColumnNames()
+        private Map<String, Map<String, Integer>> indexes;
+        protected void initColumnsAndIndices()
         {
-            if (columnNames == null)
+            if (columnNames == null || indexes == null)
             {
                 List<WebElement> headerCellElements = Locators.headerCells.findElements(this);
+                int offset = 0;
                 if (hasSelectColumn())
                 {
                     headerCellElements.remove(0);
+                    offset = 1;
                 }
                 columnNames = getWrapper().getTexts(headerCellElements);
+                indexes = new HashMap<>();
                 for (int i = 0; i < headerCellElements.size(); i++)
                 {
                     headerCells.put(columnNames.get(i), headerCellElements.get(i)); // Fill out the headerCells Map since we have them all
+                    indexes.put(columnNames.get(i), Map.of("normalizedIndex", i, "rawIndex", i + offset));
```

Comment:
Make a little [POJO](https://en.wikipedia.org/wiki/Plain_old_Java_object) for these indices instead of using magical map keys.

Reply:
done